pusher-platform-core/src/main/kotlin/com/pusher/platform/subscription/BaseSubscription.kt

```diff
@@ -58,6 +58,8 @@ internal class BaseSubscription<A>(
                     e is SSLHandshakeException -> onError(Errors.other(e))
                     else -> onError(NetworkError("Connection failed"))
                 }
+            } catch (e: Exception) {
```

Comment:
Specifically what are other expected technical failures here (besides IOException)? After having a quick look at the code I did not see any. Do we have any reference to a specific issue observed?

Definitely we should not be catching and suppressing our internal programming mistakes which we may add there in the future as those should clearly signal themselves so that they are caught and fixed as soon as introduced (the Fail-Fast principle). Catching Exception includes catching RuntimeExceptions which mean programming mistakes. We should not been passing them at the client code as there's nothing that could or should be done at that side with such internal issue. Also that would leave an illegal/undefined state and any subsequent errors would be harder to investigate as further away from the root cause.